### PR TITLE
Use relaxation to LES profiles + define initial conditions more precisely for LES_driven_SCM

### PR DESCRIPTION
**Pull request by costachris**
Addresses points 4, 7 from #505. 
Averages 1 hour window around `t_end` - 6 hours for initial conditions. `thetali` and `qt` are relaxed to the mean LES profile following eqn 9 in Shen et al. 2021

### CODE REVIEW

**Comment by ilopezgp** [comment]
This might be a good time to change the resolution to 50 m for the LES driven cases, 20 m is too small for any GCM.

**Comment by costachris** [comment]
@ilopezgp There is some kind of resolution-dependence when changing the vertical resolution from 20m to 50m (see anvil-like structures in the plot below). Increasing the `detrainment_factor` eliminates the problem, and decreasing the time step has basically no influence on these anvils. Perhaps we can merge these changes and I can address the vertical resolution in another PR? 
Screen Shot 2021-11-08 at 10 47 44 AM

File: integration_tests/utils/Cases.jl [hunk 1]
```diff
@@ -1588,14 +1588,20 @@ function CasesBase(case::LES_driven_SCM, namelist, grid::Grid, param_set, Sur, F
...
         time_interval_bool = findall(>(-t_interval_from_end_s), t_from_end_s)
         imin = time_interval_bool[1]
         imax = time_interval_bool[end]
 
-        TC.LESData(imin, imax, les_filename)
+        TC.LESData(
+            imin,
+            imax,
+            les_filename,
+            namelist["t_interval_from_end_s"],
```
Comment (ilopezgp) [inline integration_tests/utils/Cases.jl:1602]:
```suggestion
            t_interval_from_end_s,
```

File: src/Turbulence_PrognosticTKE.jl [hunk 2]
```diff
@@ -71,11 +83,20 @@ function compute_gm_tendencies!(edmf::EDMF_PrognosticTKE, grid, state, Case, gm,
...
         if force_type(Case.Fo) <: ForcingLES
             H_horz_adv = aux_gm.dTdt_hadv[k] / Π
-            H_nudge = aux_gm.dTdt_nudge[k] / Π
             H_fluc = aux_gm.dTdt_fluc[k] / Π
 
             gm_U_nudge_k = (aux_gm.u_nudge[k] - prog_gm.u[k]) / Case.Fo.nudge_tau
             gm_V_nudge_k = (aux_gm.v_nudge[k] - prog_gm.v[k]) / Case.Fo.nudge_tau
+
+            tau_k = 1.0 / compute_les_Γᵣ(grid.zc[k])
+            if tau_k != 0
```
Comment (ilopezgp) [inline src/Turbulence_PrognosticTKE.jl:92]: I think you want to check that `Γᵣ !=0` (i.e., the timescale to relaxation is not infinite), and then inside the if you compute `tau_k = 1.0 / Γᵣ`.

File: src/types.jl [hunk 3]
```diff
@@ -338,9 +338,16 @@ struct RadiationDYCOMS_RF01 end
 struct RadiationLES end
 
 Base.@kwdef mutable struct LESData
+    "Start time index of LES"
     imin::Int = 0
+    "End time index of LES"
     imax::Int = 0
+    "Path to LES stats file used to drive SCM"
     les_filename::String = nothing
+    "Drive SCM with LES data from t = [t_interval_from_end_s, end]"
```
Comment (ilopezgp) [inline src/types.jl:347]:
```suggestion
    "Drive SCM with LES data from t = [end - t_interval_from_end_s, end]"
```

**Comment by ilopezgp** [comment]
> @ilopezgp There is some kind of resolution-dependence when changing the vertical resolution from 20m to 50m (see anvil-like structures in the plot below). Increasing the `detrainment_factor` eliminates the problem, and decreasing the time step has basically no influence on these anvils. Perhaps we can merge these changes and I can address the vertical resolution in another PR?
> Screen Shot 2021-11-08 at 10 47 44 AM

I am fine with this, you can change the resolution later if there are issues that arise with it.

**Review by ilopezgp — APPROVED** [review]
LGTM

**Pull request merged**